packages/catalog-readmodel-writer/test/catalog-readmodel-writer.integration.test.ts

```diff
@@ -42,42 +67,632 @@ describe("database test", async () => {
+      );
+      expect(retrievedEservice?.metadata).toEqual({ version: 2 });
+    });
+    it("EServiceDescriptorUpdated", async () => {
+      const draftDescriptor: Descriptor = {
+        ...getMockDescriptor(),
+        interface: getMockDocument(),
+        state: descriptorState.draft,
+      };
+      const eservice: EService = {
+        ...mockEService,
+        descriptors: [draftDescriptor],
+      };
+      await writeInReadmodel(toReadModelEService(eservice), eservices, 1);
+
+      const publishedDescriptor: Descriptor = {
+        ...draftDescriptor,
+        publishedAt: new Date(),
+        state: descriptorState.published,
+      };
+      const updatedEService: EService = {
+        ...eservice,
+        descriptors: [publishedDescriptor],
+      };
+      const payload: EServiceDescriptorUpdatedV1 = {
+        eserviceId: eservice.id,
+        eserviceDescriptor: toDescriptorV1(publishedDescriptor),
+      };
+      const message: EServiceEventEnvelope = {
+        sequence_num: 1,
+        stream_id: mockEService.id,
+        version: 2,
+        type: "EServiceDescriptorUpdated",
+        event_version: 1,
+        data: payload,
+      };
+      await handleMessageV1(message, eservices);
+
+      const retrievedEservice = await eservices.findOne({
+        "data.id": mockEService.id,
+      });
+
+      expect(retrievedEservice?.data).toEqual(
+        toReadModelEService(updatedEService)
+      );
+      expect(retrievedEservice?.metadata).toEqual({ version: 2 });
+    });
+    it("EServiceRiskAnalysisDeleted", async () => {
+      const riskAnalysis = getMockValidRiskAnalysis("PA");
+      const eservice: EService = {
+        ...mockEService,
+        riskAnalysis: [riskAnalysis],
+      };
+
+      await writeInReadmodel(toReadModelEService(eservice), eservices, 1);
+
+      const updatedEService: EService = {
+        ...mockEService,
+        riskAnalysis: [],
+      };
+      const payload: EServiceRiskAnalysisDeletedV1 = {
+        eservice: toEServiceV1(updatedEService),
+        riskAnalysisId: riskAnalysis.id,
+      };
+      const message: EServiceEventEnvelope = {
+        sequence_num: 1,
+        stream_id: mockEService.id,
+        version: 2,
+        type: "EServiceRiskAnalysisDeleted",
+        event_version: 1,
+        data: payload,
+      };
+      await handleMessageV1(message, eservices);
+
+      const retrievedEservice = await eservices.findOne({
+        "data.id": mockEService.id,
+      });
+
+      expect(retrievedEservice?.data).toEqual(
+        toReadModelEService(updatedEService)
+      );
+      expect(retrievedEservice?.metadata).toEqual({ version: 2 });
+    });
   });
 });
+
+export const getMockEService = (): EService => ({
+  id: generateId(),
+  name: "eservice name",
+  description: "eservice description",
+  createdAt: new Date(),
+  producerId: generateId(),
+  technology: technology.rest,
+  descriptors: [],
+  attributes: undefined,
```

Comment:
```suggestion
```
You can skip since it's optional, plus by defining it the `.toMatchObject` won't work

Reply:
Removed